core/server/utils/gravatar.js

```diff
@@ -1,44 +1,31 @@
 var Promise = require('bluebird'),
-    config = require('../config'),
     crypto = require('crypto'),
-    https = require('https');
+    config = require('../config'),
+    request = require('../lib/request');
 
 module.exports.lookup = function lookup(userData, timeout) {
     var gravatarUrl = '//www.gravatar.com/avatar/' +
         crypto.createHash('md5').update(userData.email.toLowerCase().trim()).digest('hex') +
-        '?s=250', image;
-
-    return new Promise(function gravatarRequest(resolve) {
-        if (config.isPrivacyDisabled('useGravatar')) {
-            return resolve();
-        }
-
-        var request, timer, timerEnded = false;
-
-        request = https.get('https:' + gravatarUrl + '&d=404&r=x', function (response) {
-            clearTimeout(timer);
-
-            if (response.statusCode !== 404 && !timerEnded) {
-                gravatarUrl += '&d=mm&r=x';
-                image = gravatarUrl;
-            }
-
-            resolve({image: image});
+        '?s=250';
+
+    if (config.isPrivacyDisabled('useGravatar')) {
+        return Promise.resolve();
+    }
+
+    return request('https:' + gravatarUrl + '&d=404&r=x', {timeout: timeout || 2 * 1000})
+        .then(function () {
+            gravatarUrl += '&d=mm&r=x';
+
+            return {
+                image: gravatarUrl
+            };
+        })
+        .catch({statusCode: 404}, function () {
+            return {
+                image: undefined
+            };
+        })
+        .catch(function () {
+            // ignore error, just resolve with no image url
```

Comment:
What's the reasoning behind the double catch here? Comment seems to suggest it's the same behaviour 🤔

Reply:
I have no idea 😃. It has tests for both cases (404 and all other errors). One case just returns nothing and one case just `image:undefined`. This is how it was working before, probably wrong and can be combined into simply return no gravatar on error.